scripts/system_performance/controller/method_mapping.py

```diff
@@ -0,0 +1,218 @@
+
+
+class MethodSelectorConfigurator(object):
+    """Class containing logic to call N methods specified by user for N instruments + any additional
+     method arguments specified"""
+
+    def __init__(self):
+        # create_method_mappings
+        pass
+
+    @staticmethod
+    def create_method_mappings():
+        """A dictionary to map input, user specified methods to return equivalent method to be called
+        TODO: Create execution_time_average and run_frequency_average methods"""
+
+        return {'missing_run_numbers_report': QueryHandler().missing_run_numbers_report,
+                'execution_times': QueryHandler().execution_times,
+                # 'execution_time_average': self.execution_time_average, TODO create method
+                'run_frequency': QueryHandler().run_frequency,
+                # 'run_frequency_average': self.run_frequency_average _ TODO create method
+               }
+
+    def method_call(self, method_name, method_args):
+        """Calls user specified method and returns statistics for a given instrument"""
+        # Check input is in mapping and place method N output in instrument_dict to return
+
+        try:
+            method_output = self.create_method_mappings()[method_name](**method_args)
+        except KeyError:
+            logging.warn("Invalid Input - method '%s' does not exist try -help "
+                          "to look at existing methods and arguments", method_name)
+            method_output = None
+
+        except TypeError:
+            logging.warn("Invalid Input invalid addiitonal arguments entered "
+                         "for method: '%s'", method_name)
+            method_output = None
+        return method_output
+
+    @staticmethod
+    def get_instrument_models():
+        """Returns a list of all currently active instruments"""
+        instrument_list = query_argument_constructor.get_list_of_instruments()
+        return instrument_list
+
+    def run_every_instrument(self, instrument_dict, method_name, method_arguments=None):
+        """Returns all existing instruments in a dictionary for a given method as:
+        {instrument_name : [[method output row 1]],[[method output row 2] etc] ...}"""
+
+        for instrument in self.get_instrument_models():
+            try:
+                # if method_name in self.create_method_mappings():
+                method_arguments['instrument_id'] = int(instrument.id)
+                logging.info("Querying for instrument: {}".format(method_arguments))
+                instrument_dict[instrument[1]] = self.create_method_mappings()[method_name](**method_arguments)
+            except KeyError:
+                logging.warn("Invalid Input - method '%s' does not exist try -help "
+                                 "to look at existing methods and arguments", method_name)
+        return instrument_dict
+
+    def user_instrument_list_validate(self, instrument_input):
```

Comment:
`validate_user_instrument_list` would be better. When naming try to start with the action / functionality first followed by the variable or object being operated. e.g. `get_data` , `construct_query`